src/SBInterpolatedImage.cpp

```diff
@@ -900,6 +782,198 @@ namespace galsim {
+                which works out to
+                2 i ( kx L cos(kx L/2) - 2 sin(kx L/2)) sin (ky L/2) / kx^2 / ky.
+                Noting that kx L/2 = ikx pi, the cosines are -1^ikx and the sines are 0.
+                Of course, lim kx->0 sin(kx)/kx is 1 though, so that term survives.  Algebra
+                eventually reduces the above expression to what's in the code below.
+             */
+            double xsum(0.0), ysum(0.0);
+            for (int iky = -Ninitial/2; iky < Ninitial/2; iky++) {
+                if (iky == 0) continue;
+                ysum += std::pow(-1.0, iky) / iky * _ktab->kval(0, iky).imag();
+            }
+            for (int ikx = -Ninitial/2; ikx < Ninitial/2; ikx++) {
+                if (ikx == 0) continue;
+                xsum += std::pow(-1.0, ikx) / ikx * _ktab->kval(ikx, 0).imag();
+            }
+            xcentroid = xsum/_dk/flux;
+            ycentroid = ysum/_dk/flux;
+        }
+        return Position<double>(xcentroid, ycentroid);
+    }
+
+    std::string SBInterpolatedKImage::SBInterpolatedKImageImpl::repr() const
+    {
+        std::ostringstream oss(" ");
+        oss.precision(std::numeric_limits<double>::digits10 + 4);
+        oss << "galsim._galsim.SBInterpolatedKImage(";
+
+        oss << "galsim._galsim.ConstImageViewD(array([";
+        ConstImageView<double> reim = getRealKImage();
+        int N = _ktab->getN();
+        for (int y = 0; y<N; ++y) {
+            if (y > 0) oss <<",";
+            BaseImage<double>::const_iterator it = reim.rowBegin(y);
+            oss << "[" << *it++;
+            for (; it != reim.rowEnd(y); ++it) oss << "," << *it;
+            oss << "]";
+        }
+        oss<<"],dtype=float)), ";
+
+        oss << "galsim._galsim.ConstImageViewD(array([";
+        ConstImageView<double> imim = getImagKImage();
+        for (int y = 0; y<N; ++y) {
+            if (y > 0) oss <<",";
+            BaseImage<double>::const_iterator it = imim.rowBegin(y);
+            oss << "[" << *it++;
+            for (; it != imim.rowEnd(y); ++it) oss << "," << *it;
+            oss << "]";
+        }
+        oss<<"],dtype=float)), ";
+
+        oss << _ktab->getDk() << ", " << stepK() << ", ";
+        boost::shared_ptr<Interpolant> kinterp = getKInterp();
+        oss << "galsim.Interpolant('"<<kinterp->makeStr()<<"', "<<kinterp->getTolerance()<<"), "
+            << "galsim.GSParams("<<*gsparams<<"))";
+        return oss.str();
+    }
+
+    ConstImageView<double> SBInterpolatedKImage::SBInterpolatedKImageImpl::getRealKImage() const
+    {
+        int N = _ktab->getN();
+        tmv::Matrix<std::complex<double> > val(N,N);
+        int Nko2 = Nk/2;
+        fillKValue(val.view(), -Nko2, 1, Nko2, -Nko2, 1, Nko2);
+        ImageAlloc<double> im(N, N);
+        tmv::MatrixView<double> m(im.getData(), N, N, 1, im.getStride(), tmv::NonConj);
+        m += val.realPart();
+        return im;
+    }
```

Comment:
Yes.  This alternate constructor would only be used for the serialization.  The current constructor would still be the one that InterpolatedKImage would call.  And boost python should handle the distinction for you via the different argument signatures.